common/persistence/execution_manager.go

```diff
@@ -120,31 +120,15 @@ func (m *executionManagerImpl) UpdateWorkflowExecution(
 	request *UpdateWorkflowExecutionRequest,
 ) (*UpdateWorkflowExecutionResponse, error) {
 
-	var updateWorkflowNewEvents []*InternalAppendHistoryNodesRequest
-	var newWorkflowNewEvents []*InternalAppendHistoryNodesRequest
-	var updateWorkflowHistoryDiff, newWorkflowHistoryDiff int64 = 0, 0
-	for _, workflowEvents := range request.UpdateWorkflowEvents {
-		newEvents, err := m.serializeWorkflowEvents(request.ShardID, workflowEvents)
-		newEvents.ShardID = request.ShardID
-		if err != nil {
-			return nil, err
-		}
-		if newEvents != nil {
-			updateWorkflowNewEvents = append(updateWorkflowNewEvents, newEvents)
-			updateWorkflowHistoryDiff += int64(len(newEvents.Node.Events.Data))
-		}
+	updateWorkflowNewEvents, updateWorkflowHistoryDiff, err := m.serializeWorkflowEventBatches(request.ShardID, request.UpdateWorkflowEvents)
+	if err != nil {
+		return nil, err
 	}
 	request.UpdateWorkflowMutation.ExecutionInfo.ExecutionStats.HistorySize += updateWorkflowHistoryDiff
 
-	for _, workflowEvents := range request.NewWorkflowEvents {
-		newEvents, err := m.serializeWorkflowEvents(request.ShardID, workflowEvents)
-		if err != nil {
-			return nil, err
-		}
-		if newEvents != nil {
-			newWorkflowNewEvents = append(newWorkflowNewEvents, newEvents)
-			newWorkflowHistoryDiff += int64(len(newEvents.Node.Events.Data))
-		}
+	newWorkflowNewEvents, newWorkflowHistoryDiff, err := m.serializeWorkflowEventBatches(request.ShardID, request.NewWorkflowEvents)
```

Comment:
no nil check? already done within?

Reply:
I could do this within the nil check, but then I have to declare those variable separately. The serializeWorkflowEventBatches handles nil as it uses for in range.